### PR TITLE
Added Combine support and updated model properties to camel case (#4)

### PR DESCRIPTION
**Pull request by colinfwalsh**
Sorry for the previous PR, I should have read the Readme more thoroughly.

This might be a controversial PR, but I changed the script to automatically convert properties into camel case for the Codable models since that's the normal Swift convention.  All models have been updated to the latest version of the lemmy-js-client.

I also added Combine support in case others find that useful.

### CODE REVIEW

File: autogen/index.js [hunk 1]
```diff
@@ -140,8 +140,23 @@ function lowercaseFirstLetter(str) {
...
+                .replace('-', '')
+                .replace('_', '')
+                );
+
+    if (converted.slice(-1) === "_") {
+        return converted.slice(0, -1);
+    }
+
+    return converted
+}
```
Comment (fishcharlie) [inline autogen/index.js]: Let's keep the formatting consistent, please use tabs here.

File: Sources/Lemmy-Swift-Client/Lemmy_Swift_Client.swift [hunk 2]
```diff
@@ -82,10 +85,52 @@ public class LemmyAPI {
...
+		let regex = try? NSRegularExpression(pattern: pattern, options: [])
+		let range = NSRange(location: 0, length: count)
+		return regex?.stringByReplacingMatches(
+			in: self,
+			options: [],
+			range: range,
+			withTemplate: "$1_$2"
+		).lowercased()
+	}
+}
```
Comment (fishcharlie) [inline Sources/Lemmy-Swift-Client/Lemmy_Swift_Client.swift]: This should be in a separate file.

File: autogen/index.js [hunk 3]
```diff
@@ -140,8 +140,23 @@ function lowercaseFirstLetter(str) {
 	return str.charAt(0).toLowerCase() + str.slice(1);
 }
 
+function snakeToCamel(str) {
+    let converted = str.toLowerCase().replace(/([-_][a-z])/g, group =>
+                group
+                .toUpperCase()
+                .replace('-', '')
+                .replace('_', '')
+                );
```
Comment (fishcharlie) [inline autogen/index.js]:
```suggestion
    let converted = str.toLowerCase().replace(/([-_][a-z])/g, (group) => {
        return group.toUpperCase().replace("-", "").replace("_", "");
    });
```

**Review by fishcharlie — CHANGES_REQUESTED** [review]
What Lemmy version did you use when updating the model properties?

**Comment by colinfwalsh** [comment]
> What Lemmy version did you use when updating the model properties?

v0.19-rc19

This version will definitely need a tagged release considering all the breaking changes.

In terms of the requested changes, those all look good.  I'll commit them as soon as I'm able.

**Comment by colinfwalsh** [comment]
> > What Lemmy version did you use when updating the model properties?
> 
> v0.19-rc19
> 
> This version will definitely need a tagged release considering all the breaking changes.
> 
> In terms of the requested changes, those all look good. I'll commit them as soon as I'm able.

Changes committed - let me know if you have any other comments/suggestions.

**Comment by fishcharlie** [comment]
@colinfwalsh 

> v0.19-rc19

I don't think I wanna be using pre-release versions for this package. I'm open to conversation about this, but my initial instinct is to only use final versions.

**Comment by colinfwalsh** [comment]
> @colinfwalsh
> 
> > v0.19-rc19
> 
> I don't think I wanna be using pre-release versions for this package. I'm open to conversation about this, but my initial instinct is to only use final versions.

I think that's valid. My only counter is that it shouldn't matter since versions are tagged.  For instance, the Readme could point users to using 3.0.3 but we'd still have 3.1.0 (as an example) for the upcoming 0.19 release.  Once 0.19 is found to be stable, we update the Readme etc.  I also want to note that 0.19 is due for release fairly soon - but I understand that's a weak argument.

At the end of the day, it's up to your discretion on how you'd like to handle things.

**Comment by colinfwalsh** [comment]
I'd also like to advocate for a `develop` branch (or similar) that runs parallel to `main` for this exact scenario.

**Comment by colinfwalsh** [comment]
Since v0.19 is officially supported and all requested changes are done, can we consider merging this or would you like to test this a bit further?  I updated all the models for the official v0.19 release as well.

**Pull request closed without merging**